core/src/main/java/org/jboss/pnc/build/finder/core/BuildFinderUtils.java

```diff
@@ -374,4 +375,38 @@ private static List<String> getArchiveExtensionsFromKoji(BuildConfig config, Cli
 
         return Collections.unmodifiableList(extensionsToCheck);
     }
+
+    public Map<Checksum, Collection<String>> swapEntriesWithPreferredChecksum(
+            Map<Checksum, Collection<String>> originalMap,
+            Map<String, Collection<Checksum>> fileInverseMap,
+            ChecksumType preferredChecksumType) {
+
+        Map<Checksum, Collection<String>> preferredChecksumMap = new HashMap<>(originalMap.size(), 1.0f);
+
+        for (Map.Entry<Checksum, Collection<String>> entry : originalMap.entrySet()) {
+            Checksum checksum = entry.getKey();
+            Collection<String> files = entry.getValue();
+
+            String firstAlias = files != null ? files.stream().findFirst().orElse(null) : null;
+            if (checksum.getType().equals(preferredChecksumType) || firstAlias == null) {
+                // If the checksum type is already the preferred type or the file list is empty,
+                // there is no need to search further
+                preferredChecksumMap.put(checksum, files);
+                continue;
+            }
+
+            Collection<Checksum> fileChecksums = fileInverseMap.get(firstAlias);
```

Comment:
I think that since you take the firstAlias from a hashSet, that there is no gurante that you will always get the same object. Doesn't this affect if you are able to find it in the inverse map?

Reply:
And it's not possible that there would be 2 files with same filenames and different checksum?
Is it the full path with `!/` "filename"?

Reply:
It's ok because I need any alias of a file. The checksum map contains for every checksum all the file aliases. The inverse map contains for every file alias the list of checksums. So no matter what alias I use, I get the corresponding checksums I need. Also, the inverse map contains the full path of the filename (e.g. `foo.tar!/foo/bar.zip!/bar/myfile-1.18.0.jar`, so there is no clash possibility (same filenames and different checksums).